src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs

```diff
@@ -0,0 +1,1115 @@
+        public static Vector256<short> UnpackHigh(Vector256<short> left, Vector256<short> right) { throw null; }
+        public static Vector256<ushort> UnpackHigh(Vector256<ushort> left, Vector256<ushort> right) { throw null; }
+        public static Vector256<int> UnpackHigh(Vector256<int> left, Vector256<int> right) { throw null; }
+        public static Vector256<uint> UnpackHigh(Vector256<uint> left, Vector256<uint> right) { throw null; }
+        public static Vector256<long> UnpackHigh(Vector256<long> left, Vector256<long> right) { throw null; }
+        public static Vector256<ulong> UnpackHigh(Vector256<ulong> left, Vector256<ulong> right) { throw null; }
+        public static Vector256<sbyte> UnpackLow(Vector256<sbyte> left, Vector256<sbyte> right) { throw null; }
+        public static Vector256<byte> UnpackLow(Vector256<byte> left, Vector256<byte> right) { throw null; }
+        public static Vector256<short> UnpackLow(Vector256<short> left, Vector256<short> right) { throw null; }
+        public static Vector256<ushort> UnpackLow(Vector256<ushort> left, Vector256<ushort> right) { throw null; }
+        public static Vector256<int> UnpackLow(Vector256<int> left, Vector256<int> right) { throw null; }
+        public static Vector256<uint> UnpackLow(Vector256<uint> left, Vector256<uint> right) { throw null; }
+        public static Vector256<long> UnpackLow(Vector256<long> left, Vector256<long> right) { throw null; }
+        public static Vector256<ulong> UnpackLow(Vector256<ulong> left, Vector256<ulong> right) { throw null; }
+        public static Vector256<sbyte> Xor(Vector256<sbyte> left, Vector256<sbyte> right) { throw null; }
+        public static Vector256<byte> Xor(Vector256<byte> left, Vector256<byte> right) { throw null; }
+        public static Vector256<short> Xor(Vector256<short> left, Vector256<short> right) { throw null; }
+        public static Vector256<ushort> Xor(Vector256<ushort> left, Vector256<ushort> right) { throw null; }
+        public static Vector256<int> Xor(Vector256<int> left, Vector256<int> right) { throw null; }
+        public static Vector256<uint> Xor(Vector256<uint> left, Vector256<uint> right) { throw null; }
+        public static Vector256<long> Xor(Vector256<long> left, Vector256<long> right) { throw null; }
+        public static Vector256<ulong> Xor(Vector256<ulong> left, Vector256<ulong> right) { throw null; }
+    }
+    public static class Bmi1
+    {
+        public static bool IsSupported { get; }
+        public static uint AndNot(uint left, uint right) { throw null; }
+        public static ulong AndNot(ulong left, ulong right) { throw null; }
+        public static uint BitFieldExtract(uint value, uint start, uint length) { throw null; }
+        public static ulong BitFieldExtract(ulong value, ulong start, ulong length) { throw null; }
+        public static uint BitFieldExtract(uint value, uint contr
```

Comment:
> if on pointer with fixed it wouldn't be able to do that

Yes, you can do this. But I am not sure whether it is really needed. I think it would be fine to start with unsafe APIs only, tell folks to pin, and see where it does not work.

I am not sure whether the pining is really a problem for these sort of APIs. I won't be surprised if  folks will want to pin in a lot of cases when using these APIs anyway because of they will want the alignment.

>Also the Span versions can be in corefx

If Span versions are in corefx, they would have to be on a different type. Span is CoreLib for .NET Core, and this is .NET Core only feature, so I do not see any issues around this.